content/open-source/sigstore/policy-controller/enforce-sbom-attestation-with-policy-controller.md

```diff
@@ -0,0 +1,159 @@
+
+## Step 1 - Checking the Policy Controller is Denying Admission
+
+Before creating a `ClusterImagePolicy`, check that the Policy Controller is deployed and that your `default` namespace is labeled correctly. Run the following to check that the deployment is complete:
+
+```bash
+kubectl -n cosign-system wait --for=condition=Available deployment/policy-controller-webhook && \
+kubectl -n cosign-system wait --for=condition=Available deployment/policy-controller-policy-webhook
+```
+
+When both deployments are finished, verify the `default` namespace is using the Policy Controller:
+
+```
+kubectl get ns -l policy.sigstore.dev/include=true
+```
+
+You should receive output like the following:
+
+```
+NAME      STATUS   AGE
+default   Active   24s
+```
+
+Once you are sure that the Policy Controller is deployed and your `default` namespace is configured to use it, run a pod to make sure admission requests are handled and denied by default:
+
+```bash
+kubectl run --image gcr.io/chainguard-demo/demo-app@ test
+```
+
+Since there is no `ClusterImagePolicy` defined yet, the Policy Controller will reject the admission request due its default behavior.
+
+In the next step, you will define a policy that verifies Chainguard Images have a SBOM attestation and apply it to your cluster.
+
+## Step 2 — Creating a `ClusterImagePolicy`
+
+Now that you have the Policy Controller running in your cluster, and have the `default` namespace configured to use it, you can now define a `ClusterImagePolicy` to admit images.
+
+Open a new file with `nano` or your preferred editor:
+
+```shell
+nano /tmp/cip.yaml
+```
+
+Copy the following policy to the `/tmp/cip.yaml` file:
+
+```yaml
+# Copyright 2022 Chainguard, Inc.
+# SPDX-License-Identifier: Apache-2.0
+apiVersion: policy.sigstore.dev/v1beta1
+kind: ClusterImagePolicy
+metadata:
+  name: must-have-spdx-cue
+  annotations:
+    catalog.chainguard.dev/title: Enforce SBOM attestation
+    catalog.chainguard.dev/description: Enforce a signed SPDX SBOM attestation from a custom key
+    catalog.chainguard.dev/labels: attestation,cue
+spec:
+  images:
+    - glob: "**"
+  authorities:
+    - name: my-authority
+      keyless:
+        identities:
+          - issuer: "https://token.actions.githubusercontent.com"
+            subject: "https://github.com/chainguard-images/images/.github/workflows/release.yaml@refs/heads/main"
+      attestations:
+        - name: must-have-spdx-attestation
+          predicateType: https://spdx.dev/Document
+          policy:
+            type: cue
+            data: |
+              predicateType: "https://spdx.dev/Document"
+```
+
+The `glob: **` line, working in combination with the `authorities` and `policy` sections, will allow any image that has at least a SBOM attestation with predicate type `https://spdx.dev/Document` to be admitted into your cluster.
+
+Save the file and then apply the policy:
+
+```bash
+kubectl apply -f /tmp/cip.yaml
```

Comment:
I'm getting an error about the predicate type:

```
Error from server (BadRequest): error when creating "/tmp/cip.yaml": admission webhook "validating.clusterimagepolicy.sigstore.dev" denied the request: validation failed: invalid value: https://spdx.dev/Document: spec.authorities[0].attestations.predicateType
unsupported precicate type
```

I know in the other policies for spdx we published we used the following:

```yaml
      attestations:
        - name: must-have-spdxjson
          predicateType: spdxjson
          policy:
            type: cue
            data: |
              predicateType: "https://spdx.dev/Document"
```

Just wanted to check if it was the 97/98 lines that were causing the issue vs. the final line 102

Reply:
How did you test it ?  I assume this is related to an old version of the policy-controller.

Reply:
Try again! I made some changes to the used images, demo-app required authentication.

Reply:
I tested with the browser terminal which I am expecting most users to use in these tutorials to try it out. I spoke with @jamonation  and he is going to update as the terminal was on an older version. Thank you for the help!